framework/tests/all-systems/TypeVars.java

```diff
@@ -3,7 +3,7 @@
 
     class Test1<T> {
         void m() {
-            @SuppressWarnings("unchecked")
+            @SuppressWarnings({"unchecked", "assignment.type.incompatible"})
```

Comment:
What is the assignment error here? Is the cast type `@NonNull T`?
As x is a local variable, shouldn't the assignment always work?